xbmc/platform/win32/WIN32Util.cpp

```diff
@@ -1218,6 +1218,77 @@ bool CWIN32Util::SetThreadLocalLocale(bool enable /* = true */)
   return _configthreadlocale(param) != -1;
 }
 
+#ifndef TARGET_WINDOWS_STORE
+std::wstring CWIN32Util::GetCurrentDisplayName()
+{
+  MONITORINFOEXW mi{};
+  mi.cbSize = sizeof(mi);
+  if (!GetMonitorInfoW(MonitorFromWindow(g_hWnd, MONITOR_DEFAULTTOPRIMARY), &mi))
+    return {};
+  return mi.szDevice;
+}
+
+CWIN32Util::DisplayConfigSnapshot CWIN32Util::GetDisplayConfigSnapshot()
+{
+  uint32_t pathCount{0};
+  uint32_t modeCount{0};
+  std::vector<DISPLAYCONFIG_PATH_INFO> paths;
+  std::vector<DISPLAYCONFIG_MODE_INFO> modes;
+
+  constexpr uint32_t flags{QDC_ONLY_ACTIVE_PATHS};
+  LONG result{ERROR_SUCCESS};
+
+  do
+  {
+    if (GetDisplayConfigBufferSizes(flags, &pathCount, &modeCount) != ERROR_SUCCESS)
+      return {};
+
+    paths.resize(pathCount);
+    modes.resize(modeCount);
+
+    result = QueryDisplayConfig(flags, &pathCount, paths.data(), &modeCount, modes.data(), nullptr);
+  } while (result == ERROR_INSUFFICIENT_BUFFER);
+
+  if (result != ERROR_SUCCESS)
+    return {};
+
+  paths.resize(pathCount);
+  modes.resize(modeCount);
+
+  return {std::move(paths), std::move(modes)};
+}
+
+std::optional<CWIN32Util::DisplayConfigId> CWIN32Util::GetCurrentDisplayTargetId()
+{
+  std::wstring name{GetCurrentDisplayName()};
+  if (name.empty())
+    return std::nullopt;
+
+  return GetDisplayTargetId(name);
+}
+
+std::optional<CWIN32Util::DisplayConfigId> CWIN32Util::GetDisplayTargetId(
+    const std::wstring& gdiDeviceName)
+{
+  const CWIN32Util::DisplayConfigSnapshot snapshot{GetDisplayConfigSnapshot()};
```

Comment:
`CWIN32Util::` prefix not seems necessary

Reply:
correct, not necessary.